core/modules/util/Error.h

```diff
@@ -53,10 +53,23 @@ class Error {
      * List of known Qserv errors
      * TODO: centralize all error code (here?) see: DM-2416
      */
-    static enum {NONE=0, DUPLICATE_SELECT_EXPR} status;
+    enum {
+        NONE=0,
+        // Query plugin errors:
+        DUPLICATE_SELECT_EXPR,
+        // InfileMerger errors:
+        HEADER_IMPORT, HEADER_OVERFLOW,
+        RESULT_IMPORT, RESULT_MD5, MYSQLOPEN, MERGEWRITE, TERMINATE,
+        CREATE_TABLE,
+        MYSQLCONNECT, MYSQLEXEC, INTERNAL
+    } status;
 
-    Error(int code, std::string const& msg) :
-            code(code), msg(msg) {
+    Error() :
+        status(NONE) {
+    }
+
+    Error(int code, std::string const& msg = "") :
+        status(NONE), code(code), msg(msg) {
```

Comment:
Comment for the whole class - in general I do not like modifiable classes especially if class is simple like this one. It's easier and safer to work with objects that do not need "extended construction". This class constructors do not provide any way to initialize status, to set status to something else than NONE one has to construct it first and then modify later. You should add status parameter to non-default constructor. And I'd recommend to disallow modifications of data members entirely.